### ISSUE TITLE
how to handle `PrimaryWIndow` despawning

### ISSUE THREAD

**Issue by mockersf** [bevyengine/bevy#13698]
When despawning the `PrimaryWindow`, what should happen?
- Should it be forbidden?
- Should a new window be promoted as the new `PrimaryWindow`?
- (also true for other windows) Should camera targeting the `PrimaryWindow` be despawned?

**Comment by alice-i-cecile** [comment]
IMO, we should handle this as gracefully as possible, as long as we expose an on-by-default way to prevent the primary window from being closed without killing the app. That means promoting another window to take its place.

When windows are despawned, we should despawn any cameras targeting them. Down the line, this should be enforced through the use of relations with the "despawn when the paired entity is missing" behavior.

**Comment by tychedelia** [comment]
It would be pretty counter intuitive to me that a secondary window would be "promoted" the primary window when the first closes. We also already allow the user to configure the exit condition. Maybe I'm missing some context, but this doesn't seem like the kind of thing there is a sensible default for in multi-window scenarios. I think I am probably working with multi-window apps more than the average user, and personally I always disable primary window function on the default plugin.

**Comment by bushrat011899** [comment]
I agree that promoting an arbitrary window to become the `PrimaryWindow` doesn't seem like the right response here. In some apps it makes sense (e.g., a web browser where other windows are basically just the primary window again) but in others I could see it being very confusing (e.g., the CAD application I use spawns a second window just for a couple of tools, promoting that to the primary window seems odd).

Perhaps a window could be annotated with a `SecondaryWindow` component indicating that it is appropriate to promote that window into a `PrimaryWindow` when the primary is destroyed? If no secondary windows are available to promote, then I would expect the application to gracefully quit by default. However, that last step should be configurable for more niche use cases.
